plugins/chat/assets/javascripts/discourse/components/chat-footer.gjs

```diff
@@ -34,7 +35,9 @@ export default class ChatFooter extends Component {
             "c-footer__item"
             (if (eq this.router.currentRouteName "chat.channels") "--active")
           }}
-        />
+        >
+          <ChatFooterUnreadIndicator @messageType="channels" />
```

Comment:
One trick you can use here to avoid spreading these strings, is higher level component:

```
// plugins/chat/assets/javascripts/discourse/components/chat/footer/unread-indicator.gjs

export const ChatFooterUnreadChannelsIndicator = <template>
  <ChatFooterUnreadIndicator @messageType="channels" />
</template>

export default class ChatFooterUnreadIndicator extends {
  // ...
}
```

```
import { FooterUnreadChannelsIndicator } from "...";

<ChatFooterUnreadChannelsIndicator />
``

Reply:
Nice, I like this approach. I've also used the constants as part of this.